### DISCUSSION TITLE
Airflow email notification using AWS SES - SignatureDoesNotMatch

### DISCUSSION THREAD

**Discussion by gautamdhava** [apache/airflow#26966] [General] [answered]
Hi 
I just deployed a airflow in kubernetes. Just trying to setup an email alert during DAG failure. Tried using the Sendgrid and no luck with that. Now I'm trying to use AWS SES service to trigger emails.  Added below configs to airflow.cfg as per document and added a connection with AWS provider. I've created a new SMTP credentials from SES and using it in (ID+key) in airflow.cfg as below. But it's failing with below said error. 

[email]
email_backend = airflow.providers.amazon.aws.utils.emailer.send_email
email_conn_id = aws_default
from_email = xxx_yyy@yyy.com

[smtp]
smtp_host=email-smtp.us-west-2.amazonaws.com
smtp_starttls=True
smtp_ssl=False
smtp_user=xxxxxxxxxxxxxxxxxxxxxxxxxxx
smtp_port=587
smtp_password=xxxxxxxxxxxxxxxxxxxxxxx
smtp_mail_from=xxxx_xxxx@yyy.com

![image](https://user-images.githubusercontent.com/47755922/194850306-0d015688-8331-40c7-a4d9-39aee8b74821.png)

**DAG:**

from datetime import datetime, timedelta
from airflow import DAG
from airflow.operators.bash import BashOperator
from airflow.utils.email import send_email
from airflow.operators.email_operator import EmailOperator

def failure_function(context):
    dag_run = context.get('dag_run')
    msg = "The folder you are trying to open doesn't exist hence the task has Failed."
    subject = f"DAG {dag_run} Failed"
    send_email(to='xxx@yyyy.com', subject=subject, html_content=msg)

def success_function(context):
    dag_run = context.get('dag_run')
    msg = "Echoed Hello hence the task has executed successfully."
    subject = f"DAG {dag_run} has completed"
    send_email(to='xxxxx@yyyy.com', subject=subject, html_content=msg)

default_args = {
    'owner': 'airflow',
    'start_date': datetime(2022, 3, 21), 
}

with DAG('check_task_status_pass_fail',
        default_args=default_args,
        description='A simple job to nofity user about the execution status of their specified tasks.',
        schedule_interval='@daily',
        catchup=False
         ) as dag:

    say_hello = BashOperator(
        task_id='say_hello',
        on_failure_callback=failure_function,
        on_success_callback=success_function,
        bash_command='echo Hello' 
    )

**Error:**

botocore.exceptions.ClientError: An error occurred (SignatureDoesNotMatch) when calling the SendRawEmail operation: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.
[2022-10-10, 10:38:49 UTC] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check

**Comment by Taragolis** [comment, marked as answer]
This kind of error might refers to some issues with provided AWS Secret Access Key
- Check that your credentials don't contain whitespaces
- You try to use temporary credentials without providing AWS Secret Token

**Reply by gautamdhava** [reply]
Thank you, Taragolis .It was the issue with smtp IAM user policy. I fixed it by adding the full ses permission. Then it started working like charm.
